
from PyQt5 import QtCore, QtGui, QtWidgets

class Ui_DetailsScreen(object):

    def __init__(self, theme, list_details):
        self.defined_theme = theme
        self.list_details = list_details

    def setupUi(self, DetailsScreen):
        DetailsScreen.setObjectName("DetailsScreen")
        DetailsScreen.resize(640, 480)
        DetailsScreen.setMinimumSize(QtCore.QSize(640, 480))
        DetailsScreen.setMaximumSize(QtCore.QSize(640, 480))
        icon = QtGui.QIcon()
        icon.addPixmap(QtGui.QPixmap("Assets/details.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        DetailsScreen.setWindowIcon(icon)
        self.centralwidget = QtWidgets.QWidget(DetailsScreen)
        self.centralwidget.setStyleSheet(self.defined_theme.style_background())
        self.centralwidget.setObjectName("centralwidget")
        self.verticalLayout_2 = QtWidgets.QVBoxLayout(self.centralwidget)
        self.verticalLayout_2.setObjectName("verticalLayout_2")
        self.lbl_title = QtWidgets.QLabel(self.centralwidget)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(24)
        self.lbl_title.setFont(font)
        self.lbl_title.setAlignment(QtCore.Qt.AlignLeading|QtCore.Qt.AlignLeft|QtCore.Qt.AlignCenter)
        self.lbl_title.setObjectName("lbl_title")
        self.lbl_title.setStyleSheet(self.defined_theme.style_project_title())

        self.verticalLayout_2.addWidget(self.lbl_title)
        self.verticalLayout = QtWidgets.QVBoxLayout()
        self.verticalLayout.setObjectName("verticalLayout")
        self.horizontalLayout_2 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_2.setObjectName("horizontalLayout_2")
        self.pushButton = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton.setMinimumSize(QtCore.QSize(48, 48))
        self.pushButton.setMaximumSize(QtCore.QSize(48, 48))
        self.pushButton.setText("")
        icon1 = QtGui.QIcon()
        icon1.addPixmap(QtGui.QPixmap("Assets/projects.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.pushButton.setIcon(icon1)
        self.pushButton.setIconSize(QtCore.QSize(32, 32))
        self.pushButton.setFlat(True)
        self.pushButton.setObjectName("pushButton")
        self.horizontalLayout_2.addWidget(self.pushButton)
        self.label_2 = QtWidgets.QLabel(self.centralwidget)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.label_2.setStyleSheet(self.defined_theme.style_info())
        self.horizontalLayout_2.addWidget(self.label_2)
        self.lbl_project_id = QtWidgets.QLabel(self.centralwidget)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        self.lbl_project_id.setFont(font)
        self.lbl_project_id.setWordWrap(True)
        self.lbl_project_id.setObjectName("lbl_project_id")
        self.lbl_project_id.setStyleSheet(self.defined_theme.style_text_info())

        self.horizontalLayout_2.addWidget(self.lbl_project_id)
        spacerItem = QtWidgets.QSpacerItem(100, 20, QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Minimum)
        self.horizontalLayout_2.addItem(spacerItem)
        self.verticalLayout.addLayout(self.horizontalLayout_2)
        self.verticalLayout_2.addLayout(self.verticalLayout)
        self.horizontalLayout_3 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_3.setObjectName("horizontalLayout_3")
        self.pushButton_2 = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton_2.setMinimumSize(QtCore.QSize(48, 48))
        self.pushButton_2.setMaximumSize(QtCore.QSize(48, 48))
        self.pushButton_2.setText("")
        icon2 = QtGui.QIcon()
        icon2.addPixmap(QtGui.QPixmap("Assets/data_criação.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.pushButton_2.setIcon(icon2)
        self.pushButton_2.setIconSize(QtCore.QSize(32, 32))
        self.pushButton_2.setFlat(True)
        self.pushButton_2.setObjectName("pushButton_2")
        self.horizontalLayout_3.addWidget(self.pushButton_2)
        self.label_4 = QtWidgets.QLabel(self.centralwidget)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        self.label_4.setFont(font)
        self.label_4.setObjectName("label_4")
        self.label_4.setStyleSheet(self.defined_theme.style_info())
        self.horizontalLayout_3.addWidget(self.label_4)
        self.lbl_creation_date = QtWidgets.QLabel(self.centralwidget)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        self.lbl_creation_date.setFont(font)
        self.lbl_creation_date.setWordWrap(True)
        self.lbl_creation_date.setObjectName("lbl_creation_date")
        self.lbl_creation_date.setStyleSheet(self.defined_theme.style_text_info())
        self.horizontalLayout_3.addWidget(self.lbl_creation_date)
        spacerItem1 = QtWidgets.QSpacerItem(100, 20, QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Minimum)
        self.horizontalLayout_3.addItem(spacerItem1)
        self.verticalLayout_2.addLayout(self.horizontalLayout_3)
        self.horizontalLayout_4 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_4.setObjectName("horizontalLayout_4")
        self.pushButton_3 = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton_3.setMinimumSize(QtCore.QSize(48, 48))
        self.pushButton_3.setMaximumSize(QtCore.QSize(48, 48))
        self.pushButton_3.setText("")
        icon3 = QtGui.QIcon()
        icon3.addPixmap(QtGui.QPixmap("Assets/calendar.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.pushButton_3.setIcon(icon3)
        self.pushButton_3.setIconSize(QtCore.QSize(32, 32))
        self.pushButton_3.setFlat(True)
        self.pushButton_3.setObjectName("pushButton_3")
        self.horizontalLayout_4.addWidget(self.pushButton_3)
        self.label_6 = QtWidgets.QLabel(self.centralwidget)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        self.label_6.setFont(font)
        self.label_6.setObjectName("label_6")
        self.label_6.setStyleSheet(self.defined_theme.style_info())
        self.horizontalLayout_4.addWidget(self.label_6)
        self.lbl_movimentation_date = QtWidgets.QLabel(self.centralwidget)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        self.lbl_movimentation_date.setFont(font)
        self.lbl_movimentation_date.setWordWrap(True)
        self.lbl_movimentation_date.setObjectName("lbl_movimentation_date")
        self.lbl_movimentation_date.setStyleSheet(self.defined_theme.style_text_info())
        self.horizontalLayout_4.addWidget(self.lbl_movimentation_date)
        spacerItem2 = QtWidgets.QSpacerItem(100, 20, QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Minimum)
        self.horizontalLayout_4.addItem(spacerItem2)
        self.verticalLayout_2.addLayout(self.horizontalLayout_4)
        self.horizontalLayout_6 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_6.setObjectName("horizontalLayout_6")
        self.pushButton_5 = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton_5.setMinimumSize(QtCore.QSize(48, 48))
        self.pushButton_5.setMaximumSize(QtCore.QSize(48, 48))
        self.pushButton_5.setText("")
        icon4 = QtGui.QIcon()
        icon4.addPixmap(QtGui.QPixmap("Assets/project-management.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.pushButton_5.setIcon(icon4)
        self.pushButton_5.setIconSize(QtCore.QSize(32, 32))
        self.pushButton_5.setFlat(True)
        self.pushButton_5.setObjectName("pushButton_5")
        self.horizontalLayout_6.addWidget(self.pushButton_5)
        self.label_10 = QtWidgets.QLabel(self.centralwidget)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        self.label_10.setFont(font)
        self.label_10.setObjectName("label_10")
        self.label_10.setStyleSheet(self.defined_theme.style_info())
        self.horizontalLayout_6.addWidget(self.label_10)
        self.lbl_project_stage = QtWidgets.QLabel(self.centralwidget)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        self.lbl_project_stage.setFont(font)
        self.lbl_project_stage.setObjectName("lbl_project_stage")
        self.lbl_project_stage.setStyleSheet(self.defined_theme.style_text_info())
        self.horizontalLayout_6.addWidget(self.lbl_project_stage)
        spacerItem3 = QtWidgets.QSpacerItem(100, 20, QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Minimum)
        self.horizontalLayout_6.addItem(spacerItem3)
        self.verticalLayout_2.addLayout(self.horizontalLayout_6)
        DetailsScreen.setCentralWidget(self.centralwidget)

        self.retranslateUi(DetailsScreen)
        QtCore.QMetaObject.connectSlotsByName(DetailsScreen)

    def retranslateUi(self, DetailsScreen):
        _translate = QtCore.QCoreApplication.translate
        DetailsScreen.setWindowTitle(_translate("DetailsScreen", "MainWindow"))
        self.lbl_title.setText(_translate("DetailsScreen", "Detalhes da tarefa"))
        self.label_2.setText(_translate("DetailsScreen", "Pertence ao projeto: "))
        self.lbl_project_id.setText(_translate("DetailsScreen", f"{self.list_details[1]}"))
        self.label_4.setText(_translate("DetailsScreen", "Data de criação: "))
        self.lbl_creation_date.setText(_translate("DetailsScreen", f"{self.list_details[2]}"))
        self.label_6.setText(_translate("DetailsScreen", "Data de movimentação: "))
        self.lbl_movimentation_date.setText(_translate("DetailsScreen", f"{self.list_details[3]}"))
        self.label_10.setText(_translate("DetailsScreen", "Estagio do projeto: "))
        self.lbl_project_stage.setText(_translate("DetailsScreen", f"{self.list_details[4]}"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    DetailsScreen = QtWidgets.QMainWindow()
    ui = Ui_DetailsScreen()
    ui.setupUi(DetailsScreen)
    DetailsScreen.show()
    sys.exit(app.exec_())
